### PR TITLE
rename `ReadonlyRecord` and `ReadonlyArray` modules

### PR DESCRIPTION
**Pull request by fubhy**
I stupidly based this off of `main`, but before I re-roll this, let's merge `main` back into `next-minor` (or rebase it). This will otherwise only cause even more merge conflicts.

### CODE REVIEW

File: packages/cli/README.md [hunk 1]
```diff
@@ -151,7 +151,7 @@ Let's continue with our `minigit` example and and create the `add` and `clone` s
 
 ```ts
 import { Args, Command, Options } from "@effect/cli"
-import { Console, Option, ReadonlyArray } from "effect"
+import { Console, Option, Array as ReadonlyArray } from "effect"
```
Comment (mikearnaldi) [inline packages/cli/README.md:154]: wouldn't it be better to not re-alias?

Comment (fubhy) [reply packages/cli/README.md]: @mikearnaldi Now you can take a look ;-)

File: packages/cli/README.md [hunk 2]
```diff
@@ -209,7 +209,7 @@ Our final CLI application is as follows:
 ```ts
 import { Args, Command, Options } from "@effect/cli"
 import { NodeContext, NodeRuntime } from "@effect/platform-node"
-import { Console, Effect, Option, ReadonlyArray } from "effect"
+import { Console, Effect, Option, Array as ReadonlyArray } from "effect"
```
Comment (tim-smart) [inline packages/cli/README.md:212]: Leftover `as ReadonlyArray`

File: packages/cli/src/internal/args.ts [hunk 3]
```diff
@@ -3,14 +3,14 @@ import type * as Path from "@effect/platform/Path"
 import type * as Terminal from "@effect/platform/Terminal"
 import * as Schema from "@effect/schema/Schema"
 import * as TreeFormatter from "@effect/schema/TreeFormatter"
+import * as Array_ from "effect/Array"
```
Comment (tim-smart) [inline packages/cli/src/internal/args.ts:6]: As above.

File: packages/cli/src/internal/cliApp.ts [hunk 4]
```diff
@@ -1,13 +1,13 @@
 import type * as Terminal from "@effect/platform/Terminal"
 import * as Color from "@effect/printer-ansi/Color"
+import * as Array_ from "effect/Array"
```
Comment (tim-smart) [inline packages/cli/src/internal/cliApp.ts:3]: .

File: packages/cli/src/internal/command.ts [hunk 5]
```diff
@@ -59,7 +59,7 @@ const parseConfig = (config: Command.Command.Config): Command.Command.ParsedConf
     if (Array.isArray(value)) {
```
Comment (tim-smart) [inline packages/cli/src/internal/command.ts:59]: Add `isArray` to the `Array` module then we can rename the import.

File: packages/cli/src/internal/commandDescriptor.ts [hunk 6]
```diff
@@ -2,6 +2,7 @@ import type * as FileSystem from "@effect/platform/FileSystem"
 import type * as Path from "@effect/platform/Path"
 import type * as Terminal from "@effect/platform/Terminal"
 import * as Color from "@effect/printer-ansi/Color"
+import * as Array_ from "effect/Array"
```
Comment (tim-smart) [inline packages/cli/src/internal/commandDescriptor.ts:5]: .

File: packages/cli/src/internal/options.ts [hunk 7]
```diff
@@ -3,6 +3,7 @@ import type * as Path from "@effect/platform/Path"
 import type * as Terminal from "@effect/platform/Terminal"
 import * as Schema from "@effect/schema/Schema"
 import * as TreeFormatter from "@effect/schema/TreeFormatter"
+import * as Array_ from "effect/Array"
```
Comment (tim-smart) [inline packages/cli/src/internal/options.ts:6]: .

File: packages/effect/src/List.ts [hunk 8]
```diff
@@ -21,6 +21,7 @@
  * Licensed under Apache License 2.0
  * (http://www.apache.org/licenses/LICENSE-2.0).
  */
+import * as Array_ from "./Array.js"
```
Comment (tim-smart) [inline packages/effect/src/List.ts:24]: .

**Review by tim-smart — CHANGES_REQUESTED** [review]
In general if we add some more apis to remove the `Array_` imports, I think we can then merge this.

**Review by tim-smart — APPROVED** [review]

**Pull request merged**